### PR TITLE
add an example of customized NuGetPackageAnalyzerFunction

### PR DESCRIPTION
**Pull request by heng-liu**
This is an example of Azure function. It's based on https://nuget.visualstudio.com/NuGetMicrosoft/_git/NuGetInternal?path=%2FTools%2FNuGetPackageAnalyzerFunction%2FNuGetPackageAnalyzerFunction&version=GBmaster
The local.settings.json is in .gitignore file, so it's not uploaded.
You may check the example of local.settings.json here: (sensitive info is masked with "XXX")
```
{
  "IsEncrypted": false,
  "Values": {
    "AzureWebJobsStorage": "UseDevelopmentStorage=true",
    "FUNCTIONS_WORKER_RUNTIME": "dotnet",
    "sqldb_connectionstring": "Server=tcp:henliwindbserver.database.windows.net,1433;Initial Catalog=henliWinDB;Persist Security Info=False;User ID=adminhenli;Password=XXXXXXXX;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;",
    "ServiceBus_String": "Endpoint=sb://henliwinservicebus.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "APPINSIGHTS_INSTRUMENTATIONKEY": "101e6a67-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  }
}
```

### CODE REVIEW

File: NuGetPackageAnalyzerFunction/NuGetPackageAnalyzerFunction.csproj [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+		<None Update="host.json">
+			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+		</None>
+		<None Update="local.settings.json">
+			<CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+			<CopyToPublishDirectory>Never</CopyToPublishDirectory>
+		</None>
+	</ItemGroup>
+	<ItemGroup>
+	  <ProjectReference Include="..\..\NuGet.Client\src\NuGet.Core\NuGet.Packaging\NuGet.Packaging.csproj" />
```
Comment (loic-sharma) [inline NuGetPackageAnalyzerFunction/NuGetPackageAnalyzerFunction.csproj:24]: Can this use `PackageReference` instead?

Comment (heng-liu) [reply NuGetPackageAnalyzerFunction/NuGetPackageAnalyzerFunction.csproj:24]: Thanks for reviewing! But sorry, I can't.  
All the Azure function could only targets netcoreapp3.1 for now (as net5.0 is not GA yet).
And the NuGet doesn't support signing in netcoreapp3.1. So there is no NuGet.Packaging package available for signing related APIs for TFM netcoreapp3.1.
I have to manually change the NuGet.Client solution, to retarget it to netcoreapp3.1 and make the verification APIs  available for netcoreapp3.1 code path. 
Fortunately, the netcoreapp3.1 runtime has no difference with net5.0 in the API I'm going to use. (I remember netcroreapp3.1 runtime doesn't support signing well, but the verification works the same way with net5.0)

But if the analyzing is not relying on net5.0 runtime, we can use package reference instead. So the analyzer in this PR is a very special case.

Comment (loic-sharma) [reply NuGetPackageAnalyzerFunction/NuGetPackageAnalyzerFunction.csproj:24]: Makes sense, thanks for the info!

**Comment by nkolev92** [comment]
Don't mind me, just cleaning up. 
Feel free to reopen if need be.

**Pull request closed without merging**